src/vcastUtilities.ts

```diff
@@ -159,49 +162,71 @@ export function addIncludePath (fileUri: vscode.Uri) {
   // I'm handling a few error cases here without going crazy
   let statusMessages:string[] = [];
 
-  // read the existing file contents
   let existingJSON: any;
-  try {
-    existingJSON = JSON.parse(fs.readFileSync(fileUri.fsPath));
-    if (existingJSON.configurations.length == 0) {
-      statusMessages.push (`{configurationFile} file has no existing configurations, creating a 'vcast' configuraiton.  `); 
-      existingJSON.configurations.push ({name: "vcast", includePath: []});
-    }
-  } catch {
-    // if there is some sort of parse error with the existing file don't change it
-    vscode.window.showErrorMessage(`Exception parsing {configurationFile} file, no changes made.  Check for syntax errors.  `);
+  let existingJSONasString: string;
+  
+  // Requires json-c parsing to handle comments etc.
+  existingJSONasString  = fs.readFileSync (fileUri.fsPath).toString();
+  var parseErrors: jsonc.ParseError[] = [];  // not using programatically, for debug only
+  // note that jsonc.parse returns "real json" without the comments
+  existingJSON = jsonc.parse(existingJSONasString, parseErrors, jsoncParseOptions);
+
+  if (existingJSON && existingJSON.configurations && existingJSON.configurations.length > 0) {
+    const numberOfCofigurations = existingJSON.configurations.length;
+    statusMessages.push (`{configurationFile} file has ${numberOfCofigurations} configurations ... `); 
+  }
+  else {
+    statusMessages.push (`{configurationFile} file has no existing configurations, please add a configuration.   `); 
+    vscode.window.showErrorMessage(statusMessages.join ("\n"));
     return;
   }
-
+  
   // when we get here we should always have a configurations array
-  // but we might now have an includePath, so add it if its missing
+  // but we might not have an includePath, so add it if its missing
   
   let configName = existingJSON.configurations[0].name;
```

Comment:
This was in the old code, but why are we only doing it on `configurations[0]`?

Reply:
yes same as before, we could improve to prompt the user for the configuration they are using as a separate issue

Reply:
Okay!